docs/guides/data/creating-a-cloud-app.md

```diff
@@ -0,0 +1,329 @@
+
+Some external package managers that we can leverage when building an app include:
+
+<center>
+
+| Package Manager | Application                        |
+| --------------- | ---------------------------------- |
+| APT             | Advanced Packaging Tool for Ubuntu |
+| CPAN            | Comprehensive Perl Archive Network |
+| CRAN            | Comprehensive R Archive Network    |
+| gem             | Package Manager for Ruby           |
+| pip             | PyPI (Python Package Index)        |
+
+</center>
+
+One requirement for FastQC is that it must have a suitable [Java Runtime Environment](https://www.java.com/en/). To include this in the app, we have to edit the `dxapp.json` file. Open `dxapp.json` and append the following line to `"runSpec"`:
+
+```JSON
+  "execDepends": [
+    {"name": "openjdk-7-jre-headless",
+     "package_manager": "apt"}
+  ]
+```
+
+Now, the `"runSpec"` object should look like the following:
+
+```JSON
+  ...
+  "runSpec": {
+    "timeoutPolicy": {
+      "*": {
+        "hours": 48
+      }
+    },
+    "interpreter": "bash",
+    "release": "14.04",
+    "distribution": "Ubuntu",
+    "file": "src/dx_fastqc_example_app.sh",
+    "execDepends": [
+      {"name": "openjdk-7-jre-headless",
+       "package_manager": "apt"}
+    ]
+  },
+  ...
+```
+
+When you build and run your application, the virtual environment will now download `openjdk-7` from Ubuntu's APT package manager as a prerequisite. For more information on how to specify packages from Git, R, or Python, refer to the [Software Packages](https://wiki.dnanexus.com/Execution-Environment-Reference#Software-Packages) wiki page.
+
+## Calling FastQC
+
+The last step is to call the FastQC executable from within the app. Open up `src/dx_fastqc_example_app.sh` with a text editor. Inside this bash script is where we will be working with FastQC and our data. Before we dive in, its a good idea to add a few useful parameters for the script execution.
+
+Right after the Bash shebang (`#!/bin/bash`), add the following line:
+
+```bash
+set -e -x
+```
+
+Below is a table describing what each flag does:
+
+<center>
+
+| Flag | Description                                                 |
+| ---- | ----------------------------------------------------------- |
+| -e   | Exit immediately if a command exits with a non-zero status. |
+| -x   | Print each command to standard error before execution.      |
+
+</center>
+
+After the application downloads the input file (`dx download "$bam_file" -o bam_file`), we need to create the appropriate output directories and run FastQC on our BAM file. Add the following lines to the bash script within the `main` function:
+
+```bash
+mkdir ~/fastqc-out/                               # FastQC Output Folder
+fastqc "$bam_file_name" -o ~/fastqc-out           # Runs FastQC on BAM File
+```
+
+## Uploading Files
+
+After FastQC finishes, the last thing to do is to upload the reports generated by FastQC to our project.
```

Comment:
One thing that struck me — I don't know if we hit on enough the difference between running in a cloud environment vs. cluster. For instance, I think many people will look at this and say "why do we need to upload a report?". I do think there should be a short paragraph at the beginning stating "developing in the cloud is different because you get this transient machine that has to download the inputs + install dependencies upon startup and then upload the results back to a cloud filesystem"... Maybe use better words than I did but you get the idea.

Reply:
I added a short paragraph on the benefits/differences of writing and running a cloud app and I also expanded on why its necessary to upload any output files in the "Uploading Files" section.